src/tests/ftest/pool/management_race.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python3
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+import time
+import random
+from apricot import TestWithServers
+from thread_manager import ThreadManager
+from command_utils_base import CommandFailure
+
+class PoolManagementRace(TestWithServers):
+    """
+    Epic: Create system level tests that cover pool management race boundary tests and
+          functionality.
+    Testcase:
+          DAOS-8463: Pool management integrity with race boundary conditions
+    Test Class Description:
+          Start DAOS servers, create pools and delete the pool while query and list.
+    :avocado: recursive
+    """
+
+    def del_recreate_query_and_list_pools(
+            self, del_recreate=True, pool_num=1, thread_num=1, test_loop=2):
+        """ To delete and recreate pool with the same label, to query and list pools.
+        Args:
+            del_recreate (bool): flag to delete and recreate pool with the same label.
+            pool_num (int): pool number.
+            thread_num (int): thread number.
+            test_loop (int): number of test loop to be executed.
+        """
+
+        wait_time = self.params.get("wait_time", '/run/boundary_test/*')
+        max_query_time = self.params.get("max_query_time", '/run/boundary_test/*')
+        pool_label_prefix = self.params.get("pool_label_prefix", '/run/boundary_test/*')
+        label = pool_label_prefix + str(pool_num)
+        for test_num in range(test_loop):
+            if del_recreate:
+                destroy_result = self.get_dmg_command().pool_destroy(
+                    pool=self.pool[pool_num].label.value, force=1)
+                self.log.info(
+                    "--(%d.2.%d.%d)Pool %s deleted.\n",
+                    thread_num, pool_num, test_num, label)
+                new_pool = self.get_pool()
+                new_pool.set_property(prop_name="label", prop_value=label)
+                new_pool.label.update(label)
+                self.pool[pool_num] = new_pool
```

Comment:
```suggestion
                self.pool.append(self.get_pool(create=False))
                self.pool[-1].label.update(label)
                self.pool[-1].create()
```
Why do you create a pool first, then call `set_property` to update the label? It's more efficient to set the label you want to use first, then create.

Reply:
updated.

Reply:
Logic is choose random pool to destroy for the race condition.
Original line 46 to line 49 is to replace the new pool object to self.pool, not append to the self.pool[-1].